server/game/core/ongoingEffect/OngoingEffectEngine.ts

```diff
@@ -49,7 +49,7 @@ export class OngoingEffectEngine {
             } else {
                 const triggeringEvents = events.filter((event) => properties.when[event.name]);
                 if (triggeringEvents.length > 0) {
-                    if (!properties.multipleTrigger && effect.duration !== Duration.Persistent) {
+                    if (properties.limit.isAtMax(effect.source.owner)) {
```

Comment:
Shouldn't this check come after the limit is incremented?

Reply:
I think it could, but it would require some refactoring. I'm guessing that right now the effect is sitting around until the next regroup phase. What do you think?

Reply:
I think it would be better to make sure it gets removed as soon as the limit is hit, (a) just to keep things clean for debugging purposes and (b) to prevent any weird potential bugs due to having the triggers still sitting around